engine/suspender.go

```diff
@@ -15,90 +15,104 @@ import (
 // it will read and write namespaces' annotations, and scale resources.
 func (eng *Engine) Suspender(ctx context.Context, cs *kubernetes.Clientset) {
 	eng.Mutex.Lock()
-	eng.RunningNamespacesList = make(map[string]time.Time)
 	eng.Logger.Info().Str("routine", "suspender").Msg("suspender started")
 	eng.Mutex.Unlock()
 
+	var id int
 	for {
+		// QUESTION: Why unlocking here ?
```

Comment:
As there are multiple breakpoints in this `for{}` loop, unlocking before the `n := <-eng.Wl` operation (which is a blocking one) guarantees that we will not end in a deadlock. This way, we do not have to bother about mutexes in the code that follows